arabic/unpacked/tex.js

```diff
@@ -0,0 +1,205 @@
+MathJax.Hub.Register.StartupHook('Arabic TeX Startup', function () {
+  MathJax.Hub.Register.StartupHook('TeX Jax Ready', function () {
+    var TEX = MathJax.InputJax.TeX;
+    var texParseMMLToken = TEX.Parse.prototype.mmlToken;
+    var texParseAlignedArray = TEX.Parse.prototype.AlignedArray;
+    var dict = MathJax.Hub.config.Arabic.dict;
+
+
+    TEX.Definitions.Add({
+      macros: {
+        'ar': 'HandleArabic',
+        'alwaysar': 'MarkAsArabic',
+        'fliph': 'HandleFlipHorizontal'
+      }
+    });
+
+
+    TEX.Definitions.Add({
+      macros: function () {
+        var definitions = {};
+
+        Object.keys(dict).forEach(function (key) {
+          var texCommand = dict[key][0];
+          definitions[texCommand] = key;
+        });
+
+        return definitions;
+      }()
+    });
+
+
+    TEX.Parse.Augment(function () {
+      var parsers = {};
+
+      Object.keys(dict).forEach(function (key) {
+        parsers[key] = dict[key][1]; // Parser function
+      });
+
+      return parsers;
+    }());
+
+
+    TEX.Parse.Augment({
+      flipHorizontal: function (token) {
+        return token.With({
+          // Invert the value, because flipping twice means, it is not flipped
+          fliph: !token.Get('fliph')
+        });
+      },
+      arabicNumber: function (token) {
+        var numbersMap = MathJax.Hub.config.Arabic.numbersMap;
+        var text = token.data[0].data[0];
+        var mapped = text;
+
+        Object.keys(numbersMap).forEach(function (arabicNumber) {
+          var hindiNumber = numbersMap[arabicNumber];
+          var regex = new RegExp('' + arabicNumber, 'g');
+          mapped = mapped.replace(regex, hindiNumber);
+        });
```

Comment:
>       mapped = mapped.replace(/[0-9]/g,function (m) {return numbersMap[m]});
 - [ ] Sounds good, thanks for the suggestion. Will try that :)


> You could do something similar for the ones in arabicOperator and arabicIdentifier as well. (Though if you wanted those translations to be configurable, you would have to generate the regular expressions from the data. That can be done, with a little care, up front so that it doesn't have to be done every time an operator or identifier is processed.)

 - [ ] I'll see how can I do it with multi chars identifiers such as `tan` and `sin` for example. But definitely worth trying.